src/calcium/ca.jl

```diff
@@ -0,0 +1,1447 @@
+   C = a.parent
+   r = C()
+   ccall((:ca_div_fmpz, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpz}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function //(a::ca, b::fmpq)
+   C = a.parent
+   r = C()
+   ccall((:ca_div_fmpq, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpq}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+//(a::ca, b::qqbar) = a // parent(a)(b)
+
+function //(a::Int, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_si_div, libcalcium), Nothing,
+         (Ref{ca}, Int, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function //(a::fmpz, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_fmpz_div, libcalcium), Nothing,
+         (Ref{ca}, Ref{fmpz}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function //(a::fmpq, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_fmpq_div, libcalcium), Nothing,
+         (Ref{ca}, Ref{fmpq}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+//(a::qqbar, b::ca) = parent(b)(a) // b
+
+divexact(a::ca, b::ca) = a // b
+divexact(a::ca, b::Int) = a // b
+divexact(a::ca, b::fmpz) = a // b
+divexact(a::ca, b::fmpq) = a // b
+divexact(a::ca, b::qqbar) = a // b
+divexact(a::Int, b::ca) = a // b
+divexact(a::fmpz, b::ca) = a // b
+divexact(a::fmpq, b::ca) = a // b
+divexact(a::qqbar, b::ca) = a // b
+
+function ^(a::ca, b::ca)
+   a, b = same_parent(a, b)
+   C = a.parent
+   r = C()
+   ccall((:ca_pow, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function ^(a::ca, b::Int)
+   C = a.parent
+   r = C()
+   ccall((:ca_pow_si, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Int, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function ^(a::ca, b::fmpz)
+   C = a.parent
+   r = C()
+   ccall((:ca_pow_fmpz, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpz}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function ^(a::ca, b::fmpq)
+   C = a.parent
+   r = C()
+   ccall((:ca_pow_fmpq, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpq}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+^(a::ca, b::qqbar) = a ^ parent(a)(b)
+
+^(a::Int, b::ca) = parent(b)(a) ^ b
+^(a::fmpz, b::ca) = parent(b)(a) ^ b
+^(a::fmpq, b::ca) = parent(b)(a) ^ b
+^(a::qqbar, b::ca) = parent(b)(a) ^ b
+
+function -(a::ca)
+   C = a.parent
+   r = C()
+   ccall((:ca_neg, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, C)
+   check_special(r)
+   return r
+end
+
+function inv(a::ca)
+   C = a.parent
+   r = C()
+   ccall((:ca_inv, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, C)
+   check_special(r)
+   return r
+end
+
+###############################################################################
```

Comment:
We usually put the predicates before the arithmetic ops, usually in a section called Basic manipulation.